File: codes/ABC/ABC164/ABC164_D.js
const main = input => {
    const mod = 2019
    let s = input.trim().split('')
    let ln = s.length
    
    let ms = new Array(ln+1).fill(0)
    let cr = 1
    for(let i = ln-1; 0 <= i; i--) {
      ms[i] = (ms[i+1] + cr*Number(s[i])) % mod
      cr = (cr*10) % mod
    }
    
    let cnt = new Array(mod).fill(0)
    ms.forEach(v => cnt[v] += 1)
    let ans = 0
    cnt.forEach(v => {if(v >= 2) ans += v*(v-1)/2})
    console.log(ans.toString())
  }
  
  main(require("fs").readFileSync("/dev/stdin", "utf8"))

File: codes/ABC/ABC164/ABC164_D.py
from collections import Counter
mod = 2019
s = list(input())
ln = len(s)

ms = [0]*(ln+1)
ms[ln] = 0
cr = 1
for i in range(ln):
  ms[ln-1-i] = (ms[ln-i] + cr*int(s[ln-1-i])) % mod
  cr = (cr*10) % mod

cnts = Counter()
for i in ms:
  cnts[i] += 1
ans = 0
for k,v in cnts.items():
  if v >= 2:
    ans += int(v*(v-1)/2)
print(ans)